CMakeLists.txt

```diff
@@ -30,9 +30,11 @@ ev_setup_python_executable(
 
 if((${CMAKE_PROJECT_NAME} STREQUAL ${PROJECT_NAME} OR ${PROJECT_NAME}_BUILD_TESTING) AND BUILD_TESTING)
     set(EVEREST_FRAMEWORK_BUILD_TESTING ON)
-    evc_include(CodeCoverage)
-
-    append_coverage_compiler_flags()
+    # this policy allows us to link gcov to targets defined in other directories
+    if(POLICY CMP0079)
+        cmake_policy(SET CMP0079 NEW)
+        set(CMAKE_POLICY_DEFAULT_CMP0079 NEW)
+    endif()
```

Comment:
We shouldn't need this, as the required cmake version is already > 3.13.
Furthermore, `set(CMAKE_POLICY_DEFAULT_CMP0079 NEW)` already enforces `cmake_policy(SET CMP0079 NEW)`, so only the latter would be necessary.  I also don't get the point of `if(POLICY CMP0079)`

Reply:
I'm also very confused why this is needed, since it should be the default. However on CMake 3.31 I get a
```
Attempt to add link library "gcov" to target "manager" which is not built
  in this directory.

  This is allowed only when policy CMP0079 is set to NEW.
Call Stack (most recent call first):
  tests/CMakeLists.txt:69 (append_coverage_compiler_flags_to_target)
```
error as if that policy isn't set.
the `if(POLICY CMP0079)` just ensures that the policy is available before setting it, maybe not necessary on CMake >= 3.13